EWL-6949: A1 | Image alignment for JAMA component

russom-woldezghi opened the pull request:
<!-- NOTE: Put "N/A" for any section below that isn't applicable to the work you've done, **do not omit entirely**. Before submitting a Pull Request ensure that your work complies with the [Guidelines for Contributions](CONTRIBUTING.md) and the [SG2 Standards](ama-style-guide-2/docs/standards.md). -->

## Ticket

- [EWL-6949: A1 | Image alignment for JAMA component](https://issues.ama-assn.org/browse/EWL-6949)

## Description
- Adds ama__image-wrap class to jama articles with image
- Aligns JAMA article images to the right 

## To Test
- [x] Pull down branch run `gulp serve`
- [x] Enable local SG2 in your D8 environment
- [x] Checkout `develop` branch for `ama-one.local` and  clear cache `drush @one.local cr`
- [x] Add JAMA component using the WYSIWYG button on any news article node type. 
- [x] Within the JAMA articles (inside the JAMA component node), add differently sized (and ratio) images. 
- [x] Observe images are flush to right. 
- [x] Did you test in IE 11?

## Visual Regressions

N/A

## Relevant Screenshots/GIFs

### Before
![screen shot 2019-02-15 at 2 17 55 pm](https://user-images.githubusercontent.com/541745/52881389-6d192100-3132-11e9-803e-0ed58c3599b1.png)
------
### After
![screen shot 2019-02-15 at 2 48 34 pm](https://user-images.githubusercontent.com/541745/52881405-74402f00-3132-11e9-9461-1d5fe1f91f74.png)



## Remaining Tasks
N/A

## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

Commit:
Adds ama__image-wrap class to jama articles with image

adriansmuresan commented:
@russom-woldezghi this looks good at the desktop viewport size and the images appear as having the same size even if one is bigger than the other, but at smaller viewports, the image that is larger looks slightly bigger than the one that is smaller. It may be a non-issue, I just wanted to point that out in case it matters.

Commit:
Add width to title of JAMA articles

russom-woldezghi commented:
Thanks for catching that Adrian!

I added `width:100%` to fix the image sizes on mobile displays: 

![screen shot 2019-02-18 at 11 50 38 am](https://user-images.githubusercontent.com/541745/52965697-734a1000-3373-11e9-9cb2-56961787c734.png)

mgrahamama reviewed (approved):
Works as expected on chrome, safari, ff, and ie11. Works correctly on mobile as well.

JaeBea reviewed (changes requested):
@russom-woldezghi ,
When looking at this on a mobile version the images are moved to one side. I am assuming the images are supposed to either disappear or go full with correct?

![image](https://user-images.githubusercontent.com/4438120/53988515-43c43300-40e9-11e9-91f2-d82b22aa6d6e.png)

Commit:
Merge branch 'develop' into bugfix/EWLJ-6949-image-alignment-jama-article

Commit:
Adjust Jama article to make JAMA article image and text stacked on mobile

Commit:
Adjust article stubs on category pages on mobile, above the footer below content

russom-woldezghi commented:
@JaeBea I have fixed the mobile viewport issue. I have stacked the text/title above the image. 

I also adjusted the article stub teasers on category page. The first one was aligned, but the last two were not (adding margin-left). Link to page on local: http://ama-one.local/about/leadership/how-marshfield-clinic-health-system-moves-medicine (above the footer, below the main content). 

![Screen Shot 2019-03-14 at 4 14 16 PM](https://user-images.githubusercontent.com/541745/54389169-ddee2300-4675-11e9-98da-154349b92f28.png)

russom-woldezghi committed:
Merge branch 'develop' into bugfix/EWLJ-6949-image-alignment-jama-article

designerneil commented on styleguide/source/assets/scss/02-molecules/_jama.scss:
```diff
@@ -26,17 +26,29 @@
     }
 
     a {
+      @include breakpoint($bp-xs max-width) {
```
I have this as the default then override it with something like 
`@include breakpoint($bp-small) {
  width: auto;
}`

designerneil replied:
The rule of thumb is mobile values are default and gets overridden for bigger size viewports

russom-woldezghi replied:
Thanks @designerneil - I have updated the pr with your recommendations. Let me know, if this approach works.

Commit:
adding bp-small breakpoint, keeps layout consistent between mobile and desktop

JaeBea commented on styleguide/source/assets/scss/02-molecules/_jama.scss:
```diff
@@ -25,18 +25,23 @@
       }
+      .ama__image-wrap {
+        @include breakpoint($bp-xs max-width) {
+          margin-left: 0;
```
@russom-woldezghi ,
Could you rewrite `@include breakpoint($bp-xs max-width)`  so that is defaults to `margin-left: 0` and uses `@include breakpoint($bp-xs)` to set the margin left to auto for other screens? This looks like the only remaining change needed.

russom-woldezghi replied:
Thanks @JaeBea. I have updated the `ama__image-wrap` class to the following: 
```
.ama__image-wrap {
 @include breakpoint($bp-xs) {
  margin-left: auto;
 }
}
```

Commit:
Rewrite breakpoint for ama image wrap div on JAMA article
